arviz/tests/test_data.py

```diff
@@ -631,7 +658,15 @@ def test_io_function(self, data, eight_schools_params):
         inference_data = self.get_inference_data(  # pylint: disable=W0612
             data, eight_schools_params
         )
-        assert hasattr(inference_data, "posterior")
+        groups = inference_data._groups  # pylint: disable=protected-access

```

Comment:
Comment on what this code block is doing, in particular why there's an if else would be helpful

Reply:
I can also write all test dict by hand if it is more clear, I will only have to write it one, so it is not too much work.